mrtrix3_connectome.py

```diff
@@ -626,36 +754,38 @@ def findAtlasFile(filepath, description):
       fnirt_warp_subject2template_path = fsl.findImage('T1_to_template_warpcoef')
 
       # Use result of registration to transform atlas parcellation to subject space
-      run.command(invwarp_cmd + ' --ref=T1_registered.nii --warp=' + fnirt_warp_subject2template_path + ' --out=template_to_T1_warpcoef.nii')
+      run.command(shared.invwarp_cmd + ' --ref=T1_registered.nii --warp=' + fnirt_warp_subject2template_path + ' --out=template_to_T1_warpcoef.nii')
       file.delTemporary(fnirt_warp_subject2template_path)
       fnirt_warp_template2subject_path = fsl.findImage('template_to_T1_warpcoef')
-      run.command(applywarp_cmd + ' --ref=T1_registered.nii --in=' + parc_image_path + ' --warp=' + fnirt_warp_template2subject_path + ' --out=atlas_transformed.nii --interp=nn')
+      run.command(shared.applywarp_cmd + ' --ref=T1_registered.nii --in=' + shared.parc_image_path + ' --warp=' + fnirt_warp_template2subject_path + ' --out=atlas_transformed.nii --interp=nn')
       file.delTemporary(fnirt_warp_template2subject_path)
       transformed_atlas_path = fsl.findImage('atlas_transformed')
 
     file.delTemporary('T1_registered_histmatch.nii')
 
-    if parc_lut_file or mrtrix_lut_file:
-      assert parc_lut_file and mrtrix_lut_file
-      run.command('labelconvert ' + transformed_atlas_path + ' ' + parc_lut_file + ' ' + mrtrix_lut_file + ' parc.mif')
+    if shared.parc_lut_file and shared.mrtrix_lut_file:
+      run.command('labelconvert ' + transformed_atlas_path + ' ' + shared.parc_lut_file + ' ' + shared.mrtrix_lut_file + ' parc.mif')
     else: # Not all parcellations need to go through the labelconvert step; they may already be numbered incrementally from 1
       run.command('mrconvert ' + transformed_atlas_path + ' parc.mif -strides T1_registered.mif')
     file.delTemporary(transformed_atlas_path)
 
-  elif app.args.parcellation != 'none':
-    app.error('Unknown parcellation scheme requested: ' + app.args.parcellation)
-
   file.delTemporary('T1_registered.nii')
-  if app.args.output_verbosity > 2 and mrtrix_lut_file:
-    run.command('label2colour parc.mif parcRGB.mif -lut ' + mrtrix_lut_file)
 
+  if shared.output_verbosity > 2:
+    if shared.mrtrix_lut_file:
+      label2colour_lut_option = ' -lut ' + shared.mrtrix_lut_file
+    elif shared.parc_lut_file:
+      label2colour_lut_option = ' -lut ' + shared.parc_lut_file
+    else: # Use random colouring if no LUT available, but still generate the image
+      label2colour_lut_option = ''
+    run.command('label2colour parc.mif parcRGB.mif' + label2colour_lut_option)
```

Comment:
If `--parcellation` is `none`, the script crashes here because `parc.mif` hasn't been created.

Reply:
Sorry.. meant this as a comment, not review

Reply:
Whoops; good catch.

Reply:
Closed by cda3ae7.